modules/mobile/app/controllers/mobile/v0/community_care_providers_controller.rb

```diff
@@ -0,0 +1,100 @@
+        foodAndNutrition: %w[133V00000X 133VN1201X 133N00000X 133NN1002X],
+        podiatry: %w[213E00000X 213EG0000X 213EP1101X 213ES0131X 213ES0103X],
+        optometry: %w[152W00000X 152WC0802X],
+        audiologyRoutineExam: %w[231H00000X 237600000X 261QH0700X],
+        audiologyHearingAidSupport: %w[231H00000X 237600000X]
+      }.with_indifferent_access.freeze
+
+      def index
+        ccps = ppms_api.facility_service_locator(locator_params)
+        community_care_providers = ccps.map do |record|
+          Mobile::V0::CommunityCareProvider.new(
+            id: record.id,
+            name: record.provider_name,
+            address: Address.new(
+              street: record.address_street,
+              city: record.address_city,
+              state: record.address_state_province,
+              zip_code: record.address_postal_code
+            ),
+            distance: record.miles
+          )
+        end
+        page_records, page_meta_data = paginate(community_care_providers)
+        serialized = Mobile::V0::CommunityCareProvidersSerializer.new(page_records, page_meta_data)
+        render json: serialized, status: :ok
+      end
+
+      private
+
+      def ppms_api
+        FacilitiesApi::V1::PPMS::Client.new
+      end
+
+      def locator_params
+        specialty_codes = SPECIALTIES[params[:specialty]]
+        raise Common::Exceptions::InvalidFieldValue.new('specialty', params[:specialty]) if specialty_codes.nil?
+
+        lat, long = coordinates
+        unless lat && long
+          raise Common::Exceptions::UnprocessableEntity.new(
+            detail: 'User has no home latitude and longitude', source: self.class.to_s
+          )
+        end
+
+        {
+          latitude: lat,
+          longitude: long,
+          page: pagination_params[:page_number],
+          per_page: pagination_params[:page_size],
+          radius: RADIUS,
+          specialties: specialty_codes
+        }
+      end
+
+      def coordinates
+        if params[:facilityId]
+          facility = Mobile::FacilitiesHelper.get_facilities(Array(params[:facilityId])).first
+          [facility.lat, facility.long]
+        else
+          address = current_user.vet360_contact_info.residential_address
+          [address.latitude, address.longitude]
+        end
+      end
+
+      def pagination_params
+        @pagination_params ||= Mobile::V0::Contracts::GetPaginatedList.new.call(
+          page_number: params.dig(:page, :number) || DEFAULT_PAGE_NUMBER,
+          page_size: params.dig(:page, :size) || DEFAULT_PAGE_SIZE
+        )
+      end
+
+      def paginate(records)
+        url = request.base_url + request.path
+        page_records, page_meta_data = Mobile::PaginationHelper.paginate(
+          list: records, validated_params: pagination_params, url: url
+        )
+        # this is temporary. this has come up multiple times and we should develop a better solution
```

Comment:
The reason I didn't add a better fix for this is because I think it's unclear what we want out of the PaginationHelper and GetPaginatedList. I wrote the pagination helper because I saw the logic occurring in multiple places. But we don't want all params that are used in link formation to be in GetPaginatedList, so we keep doing this. There are a number of ways we could fix this, but I think we need to determine what we really want out of the GetPaginatedList first.